core/src/main/java/org/polypheny/db/algebra/core/Join.java

```diff
@@ -257,8 +257,9 @@ public String algCompareString() {
                 left.algCompareString() + "$" +
                 right.algCompareString() + "$" +
                 (condition != null ? condition.hashCode() : "") + "$" +
-                (joinType != null ? joinType.name() : "") + "&";
+                (joinType != null ? joinType.name() : "") + "$";
```

Comment:
Why change this? The alg compare string separates identifier parts for alg nodes on the same level using `&`'s. Thus, the alg compare string of every operator should end with a '&'. This should avoid (unlikely but for some trees 
 possible) cases where different trees have the same compare string.

Reply:
Reverted.